src/lib.rs

```diff
@@ -182,25 +190,7 @@ impl Invariant {
             pool_key.fee_tier.tick_spacing,
         )?;
 
-        self.pools.update(&pool_key, &pool)?;
-
-        self.positions.add(&caller, &position);
-
-        self.ticks.update(pool_key, lower_tick.index, lower_tick)?;
-        self.ticks.update(pool_key, upper_tick.index, upper_tick)?;
-
-        self.emit_event(InvariantEvent::PositionCreatedEvent {
-            block_timestamp: exec::block_timestamp(),
-            address: msg::source(),
-            pool_key,
-            liquidity_delta,
-            lower_tick: lower_tick.index,
-            upper_tick: upper_tick.index,
-            current_sqrt_price: pool.sqrt_price 
-        });
-
-        self
-            .transfer_tokens(&pool_key.token_x, None, &caller, &program, x.get())
+        self.transfer_tokens(&pool_key.token_x, None, &caller, &program, x.get())
             .await?;
 
         let second_transaction = self
```

Comment:
additionally, we should test all possible inconsistencies that we can check within create_position